`-Wunused-function` when using `c_no_clone`

When using `c_no_clone` with vec the declaration of the `_copy` methods are excluded:
https://github.com/stclib/STC/blob/845ca9dec58cdb24e5860d6c6b565b6a89afb21a/include/stc/vec.h#L130-L136

But this is not the case for the definitions of `_copy` and `_clone`:
https://github.com/stclib/STC/blob/845ca9dec58cdb24e5860d6c6b565b6a89afb21a/include/stc/vec.h#L266-L287

This leads to warnings such as:
```
/home/emily/Code/test/tmp/__pyccel__/tmp_lst.h:8:11: warning: ‘vec_int64_t_copy’ defined but not used [-Wunused-function]
    8 | #define T vec_int64_t
      |           ^~~~~~~~~~~
/home/emily/Code/test/tmp/__pyccel__/STC/include/stc/common.h:63:23: note: in definition of macro ‘c_JOIN0’
   63 | #define c_JOIN0(a, b) a ## b
      |                       ^
/home/emily/Code/test/tmp/__pyccel__/STC/include/stc/priv/template.h:30:25: note: in expansion of macro ‘c_JOIN’
   30 |   #define _c_MEMB(name) c_JOIN(Self, name)
      |                         ^~~~~~
/home/emily/Code/test/tmp/__pyccel__/STC/include/stc/priv/template.h:62:18: note: in expansion of macro ‘T’
   62 |   #define i_type T
      |                  ^
/home/emily/Code/test/tmp/__pyccel__/STC/include/stc/priv/template.h:78:16: note: in expansion of macro ‘i_type’
   78 |   #define Self i_type
      |                ^~~~~~
/home/emily/Code/test/tmp/__pyccel__/STC/include/stc/priv/template.h:30:32: note: in expansion of macro ‘Self’
   30 |   #define _c_MEMB(name) c_JOIN(Self, name)
      |                                ^~~~
/home/emily/Code/test/tmp/__pyccel__/STC/include/stc/vec.h:262:1: note: in expansion of macro ‘_c_MEMB’
  262 | _c_MEMB(_copy)(Self* self, const Self* other) {
      | ^~~~~~~
```

--- a/include/stc/vec.h
+++ b/include/stc/vec.h
@@ -266,27 +266,6 @@ STC_INLINE bool _c_MEMB(_eq)(const Self* self, const Self* other) {
 #if defined i_implement
 
 STC_DEF void
-_c_MEMB(_copy)(Self* self, const Self* other) {
-    if (self == other) return;
-    _c_MEMB(_clear)(self);
-    _c_MEMB(_reserve)(self, other->size);
-    self->size = other->size;
-    for (c_range(i, other->size))
-        self->data[i] = i_keyclone((other->data[i]));
-}
-
-STC_DEF Self
-_c_MEMB(_clone)(Self vec) {
-    Self out = vec, *self = &out; (void)self;
-    out.data = NULL; out.size = out.capacity = 0;
-    _c_MEMB(_reserve)(&out, vec.size);
-    out.size = vec.size;
-    for (c_range(i, vec.size))
-        out.data[i] = i_keyclone(vec.data[i]);
-    return out;
-}
-
-STC_DEF void
 _c_MEMB(_clear)(Self* self) {
     if (self->size == 0) return;
     _m_value *p = self->data + self->size;
@@ -352,6 +331,27 @@ _c_MEMB(_erase_n)(Self* self, const isize idx, const isize len) {
 }
 
 #if !defined i_no_clone
+STC_DEF void
+_c_MEMB(_copy)(Self* self, const Self* other) {
+    if (self == other) return;
+    _c_MEMB(_clear)(self);
+    _c_MEMB(_reserve)(self, other->size);
+    self->size = other->size;
+    for (c_range(i, other->size))
+        self->data[i] = i_keyclone((other->data[i]));
+}
+
+STC_DEF Self
+_c_MEMB(_clone)(Self vec) {
+    Self out = vec, *self = &out; (void)self;
+    out.data = NULL; out.size = out.capacity = 0;
+    _c_MEMB(_reserve)(&out, vec.size);
+    out.size = vec.size;
+    for (c_range(i, vec.size))
+        out.data[i] = i_keyclone(vec.data[i]);
+    return out;
+}
+
 STC_DEF _m_iter
 _c_MEMB(_copy_to)(Self* self, const isize idx,
                   const _m_value arr[], const isize n) {

